asyncpraw/models/helpers.py

```diff
@@ -167,29 +173,42 @@ def create(
             "visibility": visibility,
             "weighting_scheme": weighting_scheme,
         }
-        return self._reddit.post(
+        return await self._reddit.post(
             API_PATH["multireddit_base"], data={"model": dumps(model)}
         )
 
 
 class SubredditHelper(PRAWBase):
     """Provide a set of functions to interact with Subreddits."""
 
-    def __call__(self, display_name: str) -> Subreddit:
-        """Return a lazy instance of :class:`~.Subreddit`.
+    async def __call__(self, display_name: str, fetch: bool = True) -> Subreddit:
+        """Return an instance of :class:`~.Subreddit`.
 
         :param display_name: The name of the subreddit.
+        :param fetch: Determines if the subreddit will be fetched immediately.
+            This is used for specifying special subs like ``"all"``, ``"mod"``, or with
+            temporary filtering from (e.g., ``"all-redditdev-programming"``).
+
+        # TODO: Add code example for using fetch
         """
         lower_name = display_name.lower()
 
         if lower_name == "random":
-            return self._reddit.random_subreddit()
+            return await self._reddit.random_subreddit()
         if lower_name == "randnsfw":
-            return self._reddit.random_subreddit(nsfw=True)
-
-        return Subreddit(self._reddit, display_name=display_name)
-
-    def create(
+            return await self._reddit.random_subreddit(nsfw=True)
+        sub = Subreddit(self._reddit, display_name=display_name)
+        if "+" not in lower_name and lower_name not in ["mod", "all"] and fetch:
```

Comment:
Yes, is okay because praw will raise an error if the subreddit does not exist or if you try and do `._fetch()` a special subreddit construct (e.g., `' all-sub1-sub2-sub3'` or `'sub1+sub2+sub3'`). Since asyncpraw fetches on `.subreddit()` it will need to make sure it's not a special subreddit construct. Since special subreddit constructs and non existent subreddits both raise `NotFound`, it needs to check to see if it will throw a `NotFound`. If it does it then checks if it is a special subreddit construct, if it is not then it should be a nonexistent subreddit and it will re-raise the `NotFound` error like normal. The only difference is when the `NotFound` error occurs; see below:

#### PRAW

```py

subreddit = reddit.subreddit('all-redditdev') # `._fetch()` is not executed and no request occures
subreddit.subscribers # `._fetch()` is executed and praw makes a request here and will raise a `NotFound` error  
```
however this is valid usage:
```py
subreddit = reddit.subreddit('all-redditdev')
for comment in subreddit.comments(): # _fetch() is still not executed here
    # do stuff with comment
```
#### Async PRAW

```py

subreddit = await reddit.subreddit('all-redditdev') # `._fetch()` is executed and the `NotFound` error is ignored since it is a special subreddit construct
subreddit.subscribers # results in an `AtrributeError`
```
however this is valid usage:
```py
subreddit = await reddit.subreddit('all-redditdev') # `._fetch()` is executed and the `NotFound` error is ignored since it is a special subreddit construct
async for comment in subreddit.comments():
    # do stuff with comment

```

Reply:
@LilSpazJoekp 
It is not that I am against what the method does. I understand the modifications that had to be made to make async work.
However, I am not very comfortable with the fact that the `__call__` method does:
a) select a random subreddit based on argument
b) based on the presence/absence of "mod"/"all","+", it is performing different actions.

I think it would have been better if you took the second functionality (b) and made a new method for it

Reply:
Ah, I see what you are saying. I'll see what I can do.